bbl_screen-patch/kexec_ui/printerui/qml/InstallingPage.qml

```diff
@@ -52,18 +52,33 @@ Rectangle {
                     statusList.positionViewAtEnd();
                 }
                 if (datum.prompt_yesno) {
+                    textConfirmTimer.start()
                     dialogStack.popupDialog("TextConfirm", {
                         name: "installer yesno",
-                        text: datum.prompt_yesno,
-                        titles: [qsTr("Yes"), qsTr("No")],
+                        text: Qt.binding(function() {  return datum.prompt_yesno + qsTr(" Download beginning in %1 seconds.").arg(textConfirmTimer.timerCount)}),
+                        titles: [qsTr("Download"), qsTr("Cancel")],
+                        finished: Qt.binding(function() { if (!textConfirmTimer.running) return true}),
                         onYes: function() { DdsListener.publishJson("device/request/upgrade", JSON.stringify({ command: 'x1plus', yesno: true })); dialogStack.pop(); },
-                        onNo: function() { DdsListener.publishJson("device/request/upgrade", JSON.stringify({ command: 'x1plus', yesno: false })); dialogStack.pop(); }
+                        onNo: function() { DdsListener.publishJson("device/request/upgrade", JSON.stringify({ command: 'x1plus', yesno: false })); dialogStack.pop(); },
+                        onFinished: function() { DdsListener.publishJson("device/request/upgrade", JSON.stringify({ command: 'x1plus', yesno: true })); dialogStack.pop(); }
                     });
                 }
             }
         }
     }
-    
+    Timer {
+        id: textConfirmTimer
+        interval: 1000
+        repeat: true
+        running: false
+        property int timerCount:10
+        onTriggered: {
+            timerCount--;
+            if (timerCount <= -1){
+                textConfirmTimer.stop();
+            }
+        }
+    }
```

Comment:
Hm, wait, I disagree that we should automatically download.  The original goal of this prompt is to avoid downloading things in the case of users who are paranoid about downloading things from Bambu.  I think that the new x1plusd.ota changes should mean that this dialog more or less shouldn't pop up any more, except for during initial install -- so maybe we do not want a countdown here after all?